mypy/main.py

```diff
@@ -479,6 +524,43 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
+    if special_opts.python_executable is not None and special_opts.python_version is not None:
+        try:
+            py_exe_ver = _python_version_from_executable(special_opts.python_executable)
+            if py_exe_ver != special_opts.python_version:
+                parser.error(
+                    'Error: Python version {} did not match executable {}, got version {}.'.format(
+                        special_opts.python_version, special_opts.python_executable, py_exe_ver
+                    ))
+                sys.exit(2)
+            else:
+                options.python_version = special_opts.python_version
+                options.python_executable = special_opts.python_executable
+        except PythonExecutableInferenceError as e:
```

Comment:
This try-except should surround only `_python_version_from_executable` - everything else should go in an `else:` clause.